### PR TITLE
Better error handling in Batchman

### PR DESCRIPTION
**Pull request by peterhurford**
* Changes to v0.3.0.
* Introduces a new method, `robust_batch`, has been added that can attempt batching with automatic error-handling.  Call `robust_batch(batched_fn, ...)` where `batched_fn` is the function that you want to use after applying `batch` to it, and `...` are the arguments you wanted to pass to `batched_fn`.  The default amount of max retries is 3, but you can change this by passing `batchman.retries`.
* `stop = FALSE` in Batchman now works as originally intended to keep going with batch processing even if an error occurs midstream.  Batches that result in an error simply return NA instead.

### CODE REVIEW

File: inst/tests/test-robust-batch.R [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+test_that('Robust batch can batch an R Bomb without erroring.', {
+  b_fn <- get_expect_error_fn(trycatch = TRUE, stop = TRUE)
+  rbomb$reset()
+  expect_equal(
+    robust_batch(b_fn, c(fn1, fn1, fn1, fn1, rbomb$detonate), verbose = FALSE),
+    c(1, 1, 1, 1, 1)
+  )
+})
+
+test_that('It produces the same output as a regular bathc would', {
```
Comment (robertzk) [inline inst/tests/test-robust-batch.R]: typo

File: NEWS.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+## v0.3.0
+* A new method, `robust_batch`, has been added that can attempt batching with automatic error-handling.  Call `robust_batch(batched_fn, ...)` where `batched_fn` is the function that you want to use after applying `batch` to it, and `...` are the arguments you wanted to pass to `batched_fn`.  The default amount of max retries is 3, but you can change this by passing `batchman.retries`.
```
Comment (robertzk) [inline NEWS.md]: Awesome!

**Comment by robertzk** [comment]
Are we getting this? :D

**Comment by peterhurford** [comment]
@robertzk As soon as I can put some more time into it.  Last I was working on it, it still wasn't integrating well with some other packages.

File: R/batch.R [hunk 3]
```diff
@@ -16,8 +16,10 @@
 batch <- function(batch_fn, keys, splitting_strategy = NULL,
   combination_strategy = batchman::combine, size = 50, verbose = TRUE,
   trycatch = FALSE, stop = FALSE) {
-    if (is.batched_fn(batch_fn)) return(batch_fn)
+    if (isTRUE(attr(batch_fn, 'batched'))) return(batch_fn)
```
Comment (peterhurford) [inline R/batch.R]: This is lame -- somehow R now insists that `is.batched_fn` is no longer exported.  Can't figure out why.  Hasn't happened in other PRs.

Comment (robertzk) [inline R/batch.R]: Is it in the NAMESPACE file?

On Wed, Mar 25, 2015 at 12:45 PM, Peter Hurford <notifications@github.com>
wrote:

> In R/batch.R
> <https://github.com/peterhurford/batchman/pull/25#discussion_r27146350>:
>
> > @@ -16,8 +16,10 @@
> >  batch <- function(batch_fn, keys, splitting_strategy = NULL,
> >    combination_strategy = batchman::combine, size = 50, verbose = TRUE,
> >    trycatch = FALSE, stop = FALSE) {
> > -    if (is.batched_fn(batch_fn)) return(batch_fn)
> > +    if (isTRUE(attr(batch_fn, 'batched'))) return(batch_fn)
>
> This is lame -- somehow R now insists that is.batched_fn is no longer
> exported. Can't figure out why. Hasn't happened in other PRs.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/peterhurford/batchman/pull/25/files#r27146350>.
>



-- 
-Robert Krzyzanowski

Comment (peterhurford) [inline R/batch.R]: [Yes.](https://github.com/peterhurford/batchman/blob/better_error_handling/NAMESPACE#L8)

Comment (peterhurford) [inline R/batch.R]: Managed to figure this out.  I was calling `batch` in `testing-utils.R`, which apparently doesn't work.  I guess it tries to load prior to exporting happening.

**Pull request merged**